docs/using/syntax-optional.md

```diff
@@ -1,32 +1,202 @@
+:::{important}
+`myst_enable_extensions` replaces previous configuration options:
+`admonition_enable`, `figure_enable`, `dmath_enable`, `amsmath_enable`, `deflist_enable`, `html_img_enable`
+:::
+
+(syntax/typography)=
+
+## Typography
+
+Adding `"smartquotes"` to `myst_enable_extensions` (in the sphinx `conf.py` [configuration file](https://www.sphinx-doc.org/en/master/usage/configuration.html)) will automatically convert standard quotations to their opening/closing variants:
+
+- `'single quotes'`: 'single quotes'
+- `"double quotes"`: "double quotes"
+
+Adding `"replacements"` to `myst_enable_extensions` (in the sphinx `conf.py` [configuration file](https://www.sphinx-doc.org/en/master/usage/configuration.html)) will automatically convert some common typographic texts
+
+text  | converted
+----- | ----------
+``(c)``, ``(C)`` | (c)
+``(tm)``, ``(TM)`` | (tm)
+``(r)``, ``(R)`` | (r)
+``(p)``, ``(P)`` | (p)
+``+-`` | +-
+``...`` | ...
+``?....`` | ?....
+``!....`` | !....
+``????????`` | ????????
+``!!!!!`` | !!!!!
+``,,,`` | ,,,
+``--`` | --
+``---`` | ---
+
+(syntax/linkify)=
+
+## Linkify
+
+Adding `"linkify"` to `myst_enable_extensions` (in the sphinx `conf.py` [configuration file](https://www.sphinx-doc.org/en/master/usage/configuration.html)) will automatically identify "bare" web URLs and add hyperlinks:
+
+`www.example.com` -> www.example.com
+
+:::{important}
+This extension requires that [linkify-it-py](https://github.com/tsutsu3/linkify-it-py) is installed.
+Either directly; `pip install linkify-it-py` or *via* `pip install myst-parser[linkify]`.
+:::
+
+(syntax/substitutions)=
+
+## Substitutions
+
+Adding `"substitution"` to `myst_enable_extensions` (in the sphinx `conf.py` [configuration file](https://www.sphinx-doc.org/en/master/usage/configuration.html)) will allow you to add substitutions, added in either the `conf.py` using `myst_substitutions`:
+
+```python
+myst_substitutions = {
+  "key1": "I'm a **substitution**"
+}
+```
+
+or at the top of the file, in the front-matter section (see [this section](syntax/frontmatter)):
+
+````yaml
+---
+substitutions:
+  key1: "I'm a **substitution**"
+  key2: |
+    ```{note}
+    {{ key1 }}
+    ```
+  key3: "![fun fish](img/fun-fish.png)"
+---
+````
+
+:::{important}
+Keys in the front-matter will override ones in the `conf.py`.
+:::
 
-A special syntax for admonitions can optionally be enabled by setting `myst_admonition_enable = True` in the sphinx `conf.py` [configuration file](https://www.sphinx-doc.org/en/master/usage/configuration.html).
+You can use these substitutions inline or as blocks, and you can even nest substitutions in other substitutions:
 
-The key differences are that, instead of back-ticks, colons are used, and **the content starts as regular Markdown**.
-This has the benefit of allowing the content to be rendered correctly, when you are working in any standard Markdown editor.
-For example:
+:::{tabbed} Markdown Input
 
+```md
+Inline: {{ key1 }}
```

Comment:
One concern that I have with using double-curlies `{{` is that it might clash with users that are combining Jinja + MyST Markdown. I could imagine someone, for example, programmatically generating a markdown file with a Jinja template, and then interpolating substitution variables into that file as well. In that case there wouldn't be a way to accommodate both tools, right? I don't know how common this is so not sure if it's worth changing the `{{` (since that is so common for interpolation it makes sense to use it), but wanted to raise this

Reply:
These substitutions are literally combining Jinja with MyST Markdown now (as of abe0bed), i.e. you can do `{{ key1 | capitalize }}`. So it definitely makes sense that they use `{{` and this approach should be able to replace most of these use cases.
I could easily make the delimiters configurable in the markdown-it plugin, or conversely these delimiters are configurable by the any other Jinja based tool.

Reply:
ah interesting - that's pretty neat / powerful. So the use-case I had in mind when I made this comment was something like [the Carpentries](https://carpentries.org/). They have a set of "content templates" that (I think) use Jinja interpolation to generate the actual course material for a specific Carpentry course. The generated material itself might *also* use the interpolation that we describe here (e.g. define a variable in the `conf.py` and re-use it throughout the book, or something like that.

So a couple quick thoughts:

1. I suppose at the Jinja interpolation side, they could interpolate-in variables that had `{{` in the pasted-in content. E.g. they could have the variable `{{ mykey }}` and `mykey` would *equal* `{{ my-other-variable-in-myst-header }}`, that's ultimately what'd get pasted into the MyST Markdown.
2. Alternatively, it sounds like you're saying you could configure Jinja to treat a different delimeter (like `||` or something) to do the interpolation? I didn't realize that was possible!

Sorry if it seems like I am being nit-picky here, I don't think that this is a common use-case, but I am trying to make sure we don't set ourselves up for an anti-pattern here.

Reply:
Yeh you can choose whatever you want for the delimiters: https://jinja.palletsprojects.com/en/2.11.x/api/#jinja2.Environment

Reply:
`myst_sub_delimiters` now added (and explained in docs)